### PR TITLE
Add initial scala support

### PR DESCRIPTION
**Pull request by msdinit**
Hey, this just aims to add some support for Scala. I didn't add any tests as I see Kotlin does not have any, and the ones for Java are disabled - I can try and add some if you want me to.

I also issued a PR to the infra repo, hopefully everything's in order there too:
https://github.com/compiler-explorer/infra/pull/536

### CODE REVIEW

**Comment by partouf** [comment]
scala v3 returns a warning:
```
bad option '-Xlint:_' was ignored
1 warning found
Compiler returned: 0
```

**Comment by msdinit** [comment]
Ah, sorry, missed that lints were not implemented yet.
As there does not seem to be a date when they will be, I now only return lints for scala 2, and only deprecation warnings for scala 3

**Comment by partouf** [comment]
thanks!

**Comment by RubenRBS** [comment]
This is now live! https://godbolt.org/z/qYPsfj85v :)

**Comment by msdinit** [comment]
Amazing, thank you!

File: etc/config/scala.amazon.properties [hunk 1]
```diff
@@ -0,0 +1,26 @@
+compilers=&scala
+compilerType=scala
+versionFlag=-version
+objdumper=/opt/compiler-explorer/jdk-16.0.1/bin/javap
+instructionSet=java
+defaultCompiler=scalac2136
```
Comment (RubenRBS) [inline etc/config/scala.amazon.properties:6]: Hi! Quick question. Why was this chosen as the default, if a later version exists?

Comment (msdinit) [reply etc/config/scala.amazon.properties:6]: Hey
Scala 3.0 was only released less than 2 months ago, so I felt like it was a bit early to make it default.
Also, to make the example work in both versions it needs to be compatible with Scala 2 (so no top-level functions).
With these two bits in mind I thought that latest Scala 2 version would be the best default

Comment (RubenRBS) [reply etc/config/scala.amazon.properties:6]: Makes lots of sense, thank you! I'll leave a comment for future reference so we don't forget why :)

**Pull request merged**